app/models/katello/content_view_version.rb

```diff
@@ -240,12 +240,22 @@ def check_ready_to_promote!
     end
 
     def check_ready_to_destroy!
+      check_used_by_composite!
+
       if environments.any? && !organization.being_deleted?
         fail _("Cannot delete version while it is in environments: %s") % environments.map(&:name).join(",")
       end
       return true
     end
 
+    def check_used_by_composite!
+      if composite_content_views.any? && !organization.being_deleted?
+        fail _("Cannot delete version while it is in use by composite content views: %s") %
```

Comment:
Logic like this would be nice if it followed the validator pattern, cause I feel like that is essentially what is being done here: validate_destroyable or validate_deleteable ?